std/algorithm/comparison.d

```diff
@@ -983,8 +983,8 @@ template equal(alias pred = "a == b")
     assert(equal(a, b));
 
     // predicated: ensure that two vectors are approximately equal
-    double[] c = [ 1.005, 2, 4, 3];
-    assert(equal!approxEqual(b, c));
+    double[] c = [ 1.0000000005, 2, 4, 3];
```

Comment:
This worries me a little bit because I'd expect 1.005 to be close to 1.0 with the defaults.

Reply:
That's true for `approxEqual` but not for `isClose`. Actually, this is the reason, why I want to replace `approxEqual`: The default (`1e-2`) is much too weak in most cases (a little bit like a security belt, which doesn't work when needed). I replaced it with a default which depends on the used type, which is calculated in `CommonDefaultFor` (in `std.math`): `1e-4` for `float`, `1e-9` for `double` and `1e-10` for x87 `real`.